### PR TITLE
answer for js-reference-types-diagnostics

### PR DESCRIPTION
**Pull request by tina513**
comfort: 3
clarity: 5

I used mdn for string, array and map build-in functions. And I also referenced yesterdays lab assignment.

### CODE REVIEW

**Comment by tina513** [comment]
I didn't have time for testing

File: lib/diagnostic.js [hunk 1]
```diff
@@ -6,26 +6,54 @@
...
+const normalizeWords = function (stringArg) {
+  let wordArray = stringArg.split(' ');
+  let normalizeWord = function(word) {
+    return word.toUpperCase().replace(/\W/,'');
+  }
+  let normalizedWordArray = wordArray.map(normalizeWord);
+
+  })
+  return normalizedWordArray;
 };
```
Comment (payne-chris-r) [inline lib/diagnostic.js:19]: You have a syntax error somewhere here!

File: lib/diagnostic.js [hunk 2]
```diff
@@ -6,26 +6,54 @@
...
+  let normalizedWordArray = getNormalizedWords(stringArg);
+  for (let i = 0; i < normalizedWordArray.length; i++) {
+    if (UniqueWordsArray.includes(normalizedWordArray[i])) {
+      continue;
+    }else {
+      UniqueWordsArray.push(normalizedWordArray[i]);
+    }
+  }
+  return UniqueWordsArray;
 };
```
Comment (payne-chris-r) [inline lib/diagnostic.js:35]: You also have a syntax error here!

File: lib/diagnostic.js [hunk 3]
```diff
@@ -6,26 +6,54 @@
...
 
 // 3. Write a function that returns the count of words in a string.
 // Provid the *option* to count unique words instead of total words.
-const wordCount = function (/* */) {
-  /* your response here */
+const wordCount = function (stringArg) {
+  let count = getNormalizedWords(stringArg).length;
+  let uniCount = getUniqueWords(stringArg).length;
+  return uniCount;
 };
```
Comment (payne-chris-r) [inline lib/diagnostic.js:43]: getNormalizedWords is not defined here. :-(

File: lib/diagnostic.js [hunk 4]
```diff
@@ -6,26 +6,54 @@
...
+  let normalWordsArray = normalizeWords(stringArg);
+  uniMap = normalWordsArray.forEach(function(element) {
+    if(uniMap.has(element)) {
+      uniMap[element]++;
+    }else {
+      uniMap[element] = 1;
+    }
+  });
+  return uniMap;
 };
```
Comment (payne-chris-r) [inline lib/diagnostic.js:58]: uniMap.has is not a function

**Comment by payne-chris-r** [comment]
Tina, this is actually reeeeally good code. You're doing a phenomenal job. However, I'm struggling with your comment "I didn't have time for testing." You should be testing after _every thing you do_. It's the only way we keep ourselves sane! ;-)

**Pull request closed without merging**